tests/acceptance/conftest.py

```diff
@@ -5,127 +5,127 @@
 from cryptography.hazmat.primitives import serialization as crypto_serialization
 from cryptography.hazmat.primitives.asymmetric import rsa
 from cryptography.hazmat.backends import default_backend as crypto_default_backend
+from gitlab import Gitlab
+from gitlab.v4.objects import Group, Project, User
 
 from gitlabform.gitlab import AccessLevel, GitLab
 from tests.acceptance import (
-    get_gitlab,
+    allowed_codes,
     create_group,
     create_groups,
     delete_groups,
     create_project,
     get_random_name,
     create_users,
-    delete_users,
 )
 
 
 @pytest.fixture(scope="session")
-def gitlab() -> Generator[GitLab, None, None]:
-    gl = get_gitlab()
-    yield gl  # provide fixture value
+def gl():
+    return Gitlab(os.getenv("GITLAB_URL"), private_token=os.getenv("GITLAB_TOKEN"))
 
 
-@pytest.fixture(scope="class")
-def group_and_project(group, project) -> str:
-    return f"{group}/{project}"
+@pytest.fixture(autouse=True)
+def requires_license(gl: Gitlab, request):
+    if not request.node.get_closest_marker("requires_license"):
+        return
 
+    gitlab_license = gl.get_license()
+    if not gitlab_license or gitlab_license["expired"]:
+        pytest.skip("this test requires a GitLab license (Paid/Trial)")
 
-@pytest.fixture(scope="function")
-def group_and_project_for_function(group, project_for_function) -> str:
-    return f"{group}/{project_for_function}"
```

Comment:
Removed as the path string is always available from the fixture via `project.path_with_namespace`.